web/models/package_metadata.ex

```diff
@@ -0,0 +1,26 @@
+defmodule HexWeb.PackageMetadata do
+  use HexWeb.Web, :model
+
+  embedded_schema do
+    field :description, :string
+    field :licenses, {:array, :string}
+    field :links, :map
+    field :maintainers, {:array, :string}
+    field :contributors, {:array, :string}
+  end
+
+  @required_fields ~w(description)
+  @optional_fields ~w(licenses links maintainers)
+
+  def changeset(meta, params \\ :empty) do
+    cast(meta, params, @required_fields, @optional_fields)
+    |> validate_presence(:description)
+  end
```

Comment:
for now, I tried to only port existing validations into this {Package,Release}Metadata - I'm sure more could be added